### DISCUSSION TITLE
Getting a `System.AccessViolationException` when trying to use `ClosedRequested` event

### DISCUSSION THREAD

**Discussion by Eatham532** [unoplatform/uno#12891] [Q&A]
Hey trying to use the ClosedRequested event but am getting a `System.AccessViolationException`. This only happens when trying to build for windows. It works when building for GTK.

`System.AccessViolationException: 'Attempted to read or write protected memory. This is often an indication that other memory is corrupt.'`

Followed this from the wiki: https://platform.uno/docs/articles/features/app-close-handler.html

Double checked that I added `<rescap:Capability Name="confirmAppClose"/>` to the `package.appxmanifest`

I do recal it working before. Probably when I was not using the `Uno.Extensions.Navigation` and instead was using frame navigation.

**Comment by jeromelaban** [comment]
Thanks for the report. This particular API is not available on WinAppSDK, and you'll need to use the [`AppWindow` APIs instead](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/windowing/windowing-overview) for windows only.

**Comment by Eatham532** [comment]
I added a handler function to the closing event like this within the OnLaunched method
`MainWindow.AppWindow.Closing += AppWindowOnClosing;`

Using that I can cancel the window from closing. However it seems I get an exception when trying to show a MessageDialog.

`System.Runtime.InteropServices.COMException: 'Invalid window handle. (0x80070578)
Consider WindowNative, InitializeWithWindow
See https://aka.ms/cswinrt/interop#windows-sdk'`
